unified-installer/astra-unified-installer.sh

```diff
@@ -2659,41 +2802,50 @@ step_apply_resources() {
     logdebug "apply operator resources"
     local output=""
     local captured_err=""
-    if ! is_dry_run; then
-        # apply trident-acp secret if it exists
-        if [ -e "$__GENERATED_TRIDENT_ACP_SECRET_FILE" ]; then
-            output="$(kubectl apply -f "$__GENERATED_TRIDENT_ACP_SECRET_FILE" -n "$(get_trident_namespace)" 2> "$__ERR_FILE")"
-            captured_err="$(get_captured_err)"
-            if [ -z "$output" ] || [ -n "$captured_err" ]; then
-                add_problem "Failed to apply ACS registry secret: $captured_err"
-            fi
-            logdebug "$output"
-        fi
-
-        output="$(kubectl apply -k "$operators_dir" 2> "$__ERR_FILE")"
-        captured_err="$(get_captured_err)"
-
-        if [ -n "$captured_err" ]; then
-            while IFS= read -r line; do
-                if echo "$line" | grep -q "Warning:"; then
-                    logdebug "captured warning when applying kustomize resources:${__NEWLINE}$captured_err"
-                elif echo "$line" | grep -q "no objects passed to apply"; then
-                    logdebug "no kustomize resources to apply, skipping"
-                elif [ -z "$output" ] || [ -n "$line" ]; then
-                    add_problem "Failed to apply kustomize resources: $line"
-                fi
-            done < <(echo "$captured_err")
+    local -a other_args=()
+    is_dry_run && other_args+=("--dry-run=client")
```

Comment:
Main change here is replacing the `if ! is_dry_run` check with `--dry-run=client`. This allows us to test the kubectl apply commands in some capacity instead of skipping them during a dry run.